functions/src/logic/buildQueue/cleaner.ts

```diff
@@ -42,34 +42,30 @@ export class Cleaner {
       // @see https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration
       const ONE_HOUR = 1000 * 60 * 60;
       const sixHoursAgo = new Date().getTime() - 6 * ONE_HOUR;
-
-      const { seconds, nanoseconds } = lastBuildStart;
-      const buildStart = new Timestamp(seconds, nanoseconds).toDate().getTime();
+      const buildStart = new Date(lastBuildStart.seconds * 1000).getTime();
 
       if (buildStart < sixHoursAgo) {
-        const image = Dockerhub.fetchImageData(imageType, buildId);
-        if (!image) {
-          await Discord.sendDebug(
-            `[Debug] image for "started" build "${buildId}" not found, would set it to failed`,
-          );
+        const response = await Dockerhub.fetchImageData(imageType, tag);
+
+        // Image does not exist
+        if (!response) {
+          const markAsFailedMessage = `[Cleaner] Build for "${tag}" with status "started" never reported back in. Marking it as failed. It will retry automatically.`;
+          await Discord.sendAlert(markAsFailedMessage);
+          await CiBuilds.markBuildAsFailed(buildId, { reason: markAsFailedMessage });
 
-          // await Discord.sendAlert(
-          //   `[Cleaner] Build "${buildId}" with status "started" never reported back in. Marking as as failed.`,
-          // );
-          // await CiBuilds.markBuildAsFailed(buildId, { reason: 'Runner timed out or failed to report back.' });
           continue;
         }
 
-        await Discord.sendDebug(`[Debug] image "started" build  "${buildId}" found, would set it to published
-          using the following data: ${JSON.stringify(image, null, 2)}.`);
-
-        // await CiBuilds.markBuildAsPublished(buildId, {
-        //   // imageRepo: string;
-        //   // imageName: string;
-        //   // friendlyTag: string;
-        //   // specificTag: string;
-        //   // digest: string;
-        // })
+        // Image exists
+        const markAsSuccessfulMessage = `[Cleaner] Build for "${tag}" got stuck. But the image was successfully uploaded. Marking it as published.`;
+        await Discord.sendDebug(markAsSuccessfulMessage);
+        await CiBuilds.markBuildAsPublished(buildId, jobId, {
+          digest: '', // missing
```

Comment:
I assume we cannot find the digest here when the build got stuck. Can this cause troubles later on? Is there a way to find it?

Reply:
Indeed. The dockerhub v1 api only returns 200 status code vs non-200 when checking the tag for existence. 

In a worst case scenario the UI might temporarily break in the documentation website.
In case we're ever going to use the digest for some kind of verification we'd need to build something that syncs it every now and then.